lib/Sema/CSSyntacticElement.cpp

```diff
@@ -969,7 +969,7 @@ class SyntacticElementConstraintGenerator
     auto throwLoc = throwStmt->getThrowLoc();
     Type errorType;
     if (auto catchNode = ASTScope::lookupCatchNode(module, throwLoc))
-      errorType = catchNode.getExplicitCaughtType(cs.getASTContext());
+      errorType = cs.getExplicitCaughtErrorType(catchNode);
```

Comment:
My only note here is [use of DC](https://github.com/swiftlang/swift/pull/80248/files#diff-d57f6f2d466f1f207cdf2dbd0396a1dbb2caba7b6828ebdb9954948f0ac2b499R520) in existing code. Maybe we should propagate `dc` down explicitly to avoid problems in the future because `DC` in ConstraintSystem during solving could be difference from DC during constraint generation unfortunately...

Reply:
Yup, fortunately in this case being in a closure doesn't affect the result of mapping into context, but yeah I agree we ought to be passing the DeclContext down